games/datingsim/master.py

```diff
@@ -696,7 +695,8 @@ def compute_scores(self, episode_interactions: Dict) -> None:
             turn_score = {
                 "last_message": None,
                 "request_count": 0,
-                "violated_request_count": 0,
+                "violated_request_count_pattern": 0,
```

Comment:
in overall the PR looks good to me. but if we want to track ""violated_request_count_pattern" and ""violated_request_count_token_length" in scores.json, you can add saving those action["type"]s  as in line 716

like if action["type"] == "invalid format: pattern": turn_score["violated_request_count_pattern"] = 1
plus turn_score["last_message"] = action["content"] would be helpful for analyzing the errors